osc/osc_test.go

```diff
@@ -333,46 +340,75 @@ func TestReadPaddedString(t *testing.T) {
 }
 
 func TestWritePaddedString(t *testing.T) {
-	buf := []byte{}
-	bytesBuffer := bytes.NewBuffer(buf)
-	testString := "testString"
-	expectedNumberOfWrittenBytes := len(testString) + padBytesNeeded(len(testString))
-
-	n, err := writePaddedString(testString, bytesBuffer)
-	if err != nil {
-		t.Errorf(err.Error())
-	}
+	for _, tt := range []struct {
+		s   string // string
+		buf []byte // resulting buffer
+		n   int    // bytes expected
+	}{
+		{"testString", []byte{'t', 'e', 's', 't', 'S', 't', 'r', 'i', 'n', 'g', 0, 0}, 12},
+		{"testers", []byte{'t', 'e', 's', 't', 'e', 'r', 's', 0}, 8},
+		{"tests", []byte{'t', 'e', 's', 't', 's', 0, 0, 0}, 8},
+		{"test", []byte{'t', 'e', 's', 't', 0, 0, 0, 0}, 8},
+		{"tes", []byte{'t', 'e', 's', 0}, 4},
+		{"tes\0", []byte{'t', 'e', 's', 0}, 4}, // Don't add a second null terminator if one is already present
```

Comment:
these two appear to be the same thing, perhaps the first one shouldn't have the null byte?

Reply:
Input is on the left, so only line 353 feed in a pre-existing null (assuming Go strings don't normally have them), but both should result in identical output.

Reply:
ah, yes, that makes sense.